egs/tedlium/s5/local/chain/run_tdnn.sh

```diff
@@ -0,0 +1,216 @@
+pool_lpfilter_width=
+self_repair_scale=0.00001
+# training options
+num_epochs=4
+initial_effective_lrate=0.001
+final_effective_lrate=0.0001
+leftmost_questions_truncate=-1
+max_param_change=2.0
+final_layer_normalize_target=0.5
+num_jobs_initial=3
+num_jobs_final=8
+minibatch_size=128
+relu_dim=425
+frames_per_eg=150
+remove_egs=false
+common_egs_dir=
+xent_regularize=0.1
+
+
+# End configuration section.
+echo "$0 $@"  # Print the command line for logging
+
+. cmd.sh
+. ./path.sh
+. ./utils/parse_options.sh
+
+if ! cuda-compiled; then
+  cat <<EOF && exit 1
+This script is intended to be used with GPUs but you have not compiled Kaldi with CUDA
+If you want to use GPUs (and have them), go to src/, and configure and make on a machine
+where "nvcc" is installed.
+EOF
+fi
+
+# The iVector-extraction and feature-dumping parts are the same as the standard
+# nnet3 setup, and you can skip them by setting "--stage 8" if you have already
+# run those things.
+
+suffix=
+if [ "$speed_perturb" == "true" ]; then
+  suffix=_sp
+fi
+
+dir=${dir}$suffix
+train_set=train$suffix
+gmm=tri3
+gmm_dir=exp/$gmm
+ali_dir=exp/${gmm}_ali$suffix
+lats_dir=${ali_dir/ali/lats}
+treedir=exp/chain/tri3_tree$suffix
+lang=data/lang_chain
+
+mkdir -p $dir
+# if we are using the speed-perturbed data we need to generate
+# alignments for it.
+local/nnet3/run_ivector_common.sh --stage $stage \
+  --speed-perturb $speed_perturb \
+  --generate-alignments $speed_perturb || exit 1;
+
+if [ $stage -le 9 ]; then
+  # Get the alignments as lattices (gives the CTC training more freedom).
+  # use the same num-jobs as the alignments
+  nj=$(cat ${ali_dir}/num_jobs) || exit 1;
+  steps/align_fmllr_lats.sh --nj $nj --cmd "$train_cmd" data/$train_set \
+    data/lang $gmm_dir $lats_dir
+  rm ${lats_dir}/fsts.*.gz # save space
+fi
+
+if [ $stage -le 10 ]; then
+  # Create a version of the lang/ directory that has one state per phone in the
+  # topo file. [note, it really has two states.. the first one is only repeated
+  # once, the second one has zero or more repeats.]
+  rm -rf $lang
+  cp -r data/lang $lang
+  silphonelist=$(cat $lang/phones/silence.csl) || exit 1;
+  nonsilphonelist=$(cat $lang/phones/nonsilence.csl) || exit 1;
+  # Use our special topology... note that later on may have to tune this
+  # topology.
+  steps/nnet3/chain/gen_topo.py $nonsilphonelist $silphonelist >$lang/topo
+fi
+
+if [ $stage -le 11 ]; then
+  # Build a tree using our new topology.
+  steps/nnet3/chain/build_tree.sh --frame-subsampling-factor 3 \
+      --leftmost-questions-truncate $leftmost_questions_truncate \
+      --cmd "$train_cmd" 4200 data/$train_set $lang $ali_dir $treedir
+fi
+
+if [ $stage -le 12 ]; then
+  echo "$0: creating neural net configs";
+  if [ ! -z "$relu_dim" ]; then
+    dim_opts="--relu-dim $relu_dim"
+  else
+    dim_opts="--pnorm-input-dim $pnorm_input_dim --pnorm-output-dim  $pnorm_output_dim"
+  fi
+
+  # create the config files for nnet initialization
+  pool_opts=
```

Comment:
vijay, perhaps you could create an issue to remove the pooling from other
scripts, if you're not planning to make more use of that?

On Sat, Apr 2, 2016 at 5:41 PM, Vijayaditya Peddinti <
notifications@github.com> wrote:

> In egs/tedlium/s5/local/chain/run_tdnn.sh
> <https://github.com/kaldi-asr/kaldi/pull/644#discussion_r58300350>:
>
> > +  # Build a tree using our new topology.
> > +  steps/nnet3/chain/build_tree.sh --frame-subsampling-factor 3 \
> > +      --leftmost-questions-truncate $leftmost_questions_truncate \
> > +      --cmd "$train_cmd" 4200 data/$train_set $lang $ali_dir $treedir
> > +fi
> > +
> > +if [ $stage -le 12 ]; then
> > +  echo "$0: creating neural net configs";
> > +  if [ ! -z "$relu_dim" ]; then
> > +    dim_opts="--relu-dim $relu_dim"
> > +  else
> > +    dim_opts="--pnorm-input-dim $pnorm_input_dim --pnorm-output-dim  $pnorm_output_dim"
> > +  fi
> > +
> > +  # create the config files for nnet initialization
> > +  pool_opts=
>
> We are not using pooling any more. You can remove these options.
>
> —
> You are receiving this because you commented.
> Reply to this email directly or view it on GitHub
> <https://github.com/kaldi-asr/kaldi/pull/644/files/82caa06f43641485acb81b204e02eea21e5bbbeb#r58300350>
>